nyuki/transform.py

```
@@ -26,149 +29,87 @@ def get(mcs):
-                            obj=rule_cls))
-        self._rule_cls = rule_cls
-        self._rules = _TypedList(rule_cls, rules or list())
-
-    @property
-    def type(self):
-        return self._rule_cls.TYPENAME
-
-    @property
-    def rules(self):
-        return self._rules
+    def __init__(self, conditions, type=None):
+        self._conditions = conditions
 
-    @classmethod
-    def from_dict(cls, config):
+    def _clean_condition(self, condition, data):
         """
-        Create a `Ruler` object from dict 'config'. The dict must look like the
-        following:
-            {
-                "type": <rule-type-name>,       # lower-case
-                "rules": [
-                    {"fieldname": <name>, ...}, # + rule-type dependent items
-                    {"fieldname": <name>, ...},
-                    ...
-                ]
-            }
+        Format the condition string:
+        nb: strings should be {some_string!r} (includes '')
         """
-        rule_cls = _RegisteredRules.get()[config['type']]
-        rules = []
-        for params in config['rules']:
-            kp = dict(config.get('global_params', {}))
-            kp.update(params)
-            rules.append(rule_cls(**kp))
-        return cls(rule_cls, rules=rules)
+        # Unknown fields will be converted to None
+        format_fields = defaultdict(lambda: None)
+        format_fields.update(data)
+        return condition.format(**format_fields)
 
     def apply(self, data):
         """
-        Apply each rule sequentially to `data` (in-place update)
+        Apply conditions depending on the length of the condition array,
+        as it must always start with 'if' and end with 'elif', 'else'
+        or nothing.
         """
-        for rule in self.rules:
-            rule.apply(data)
-
-
-class _Rule(metaclass=_RegisteredRules):
+        for cond in self._conditions:
+            # If type 'else', apply it and leave
```

A condition block that starts with an else is not valid. You should raise an exception in that case.